core/trace/src/main/scala/org/typelevel/otel4s/trace/SpanOps.scala

```diff
@@ -167,5 +173,115 @@ object SpanOps {
       */
     def apply[F[_]](span: Span[F], trace: F ~> F): Res[F] =
       Impl(span, trace)
+
+    implicit final class SpanOpsResSyntax[F[_]](
+        private val res: Res[F]
+    ) extends AnyVal {
+
+      def translate[G[_]](fk: F ~> G, gk: G ~> F): Res[G] =
+        new Res[G] {
+          def span: Span[G] = res.span.mapK(fk)
+          def trace: G ~> G = res.trace.andThen(fk).compose(gk)
+        }
```

Comment:
is there a reason not to use `Res.apply` for this?

Reply:
`trace` at least should probably be a `val` because if you don't use it, nothing ends up traced, so it will be used in almost all cases

Reply:
Yeah, `Res.apply` fits better